social_engagement/models.py

```diff
@@ -45,6 +56,18 @@ def get_user_engagement_score(cls, course_key, user_id):
 
         return entry.score
 
+    @classmethod
+    def get_user_engagements_stats(cls, course_key, user_id):
```

Comment:
Yes that's basically the idea. It would allow to remove `_get_user_discussion_metrics` on the api-integration side, so we can just call `StudentSocialEngagementScore.get_user_engagements_stats(course_key, user.id)` directly. This is not strictly required, it'd just be nice to have.

Reply:
```python
@classmethod
def get_user_engagements_stats(cls, course_key, user_id, default=None):
    """
    Returns user's statistics as a dictionary.
    If record does not exist, it returns `default` if specified
    or else a dictionary containing statistics with their default values.
    """
    try:
        return cls.objects.get(course_id__exact=course_key, user__id=user_id).stats
    except cls.DoesNotExist:
        if default is not None:
            return default

        return {
            stat.name: stat.default
            for stat in cls._meta.fields
            if stat.name.startswith('num_')
        }
```

Does it look good?

Reply:
Oh that's nice, yep :)